src/core/sys/posix/ucontext.d

```diff
@@ -921,19 +921,125 @@ else version (CRuntime_Musl)
 }
 else version (Darwin)
 {
-    alias mcontext_t = void;
-
-    struct ucontext
+    private
     {
-        int         uc_onstack;
-        sigset_t    uc_sigmask;
-        stack_t     uc_stack;
-        ucontext*   uc_link;
-        size_t      uc_mcsize;
-        mcontext_t* uc_mcontext;
+        version (X86_Any)
+        {
+            struct __darwin_mmst_reg
+            {
+                char[10] __mmst_reg;
+                char[6] __mmst_rsrv;
+            }
+
+            struct __darwin_xmm_reg
+            {
+                char[16] __xmm_reg;
+            }
+
+            struct __darwin_exception_state
+            {
+                uint __trapno;
+                uint __err;
```

Comment:
You're missing a field here [1].

[1] https://github.com/apple/darwin-xnu/blob/0a798f6738bc1db01281fc08ae024145e84df927/osfmk/mach/i386/_structs.h#L1153

Reply:
That doesn't exist on my system [1].

[1] https://github.com/apple/darwin-xnu/blob/b472f0612b8556cd1c6eb1c285ec1953de759e35/osfmk/mach/i386/_structs.h#L751-L753

Reply:
(Though I'll note that this is inconsequential as thankfully the size doesn't change between older and newer OS X versions).

Reply:
I think you either should follow the headers or just do something like `ubyte[12] __hidden` to get the correct size.

Reply:
But that ignores the alignment. Packing repeat fields isn't all that much bad.